chain/gen/mining.go

```diff
@@ -27,7 +27,17 @@ func MinerCreateBlock(ctx context.Context, sm *stmgr.StateManager, w api.WalletA
 		return nil, xerrors.Errorf("failed to load tipset state: %w", err)
 	}
 
-	worker, err := stmgr.GetMinerWorkerRaw(ctx, sm, st, bt.Miner)
+	lbts, err := stmgr.GetLookbackTipSetForRound(ctx, sm, pts, bt.Epoch)
+	if err != nil {
+		return nil, xerrors.Errorf("getting lookback miner actor state: %w", err)
+	}
+
+	lbst, _, err := sm.TipSetState(ctx, lbts)
```

Comment:
This computes the state, we could just load it (we may also be doing this in GetMiningBase)

Reply:
Ah, I see. I should be walking back one fewer blocks and looking at the parent state?

Reply:
just do `GetTipsetByHeight` for the lookback height, and ensure `prev` is false, then take that tipsets epoch, add one, and call `GetTipsetByHeight` again on that with prev also equal to false.

Reply:
Given the null blocks, I'm pretty sure you'd end up with the exact same result.

Reply:
oh wait, `GetLookbackTipSetForRound` uses prev=true. This becomes even easier then, just call `GetTipsetByHeight(lookbackEpoch+1, false)`

Reply:
> Given the null blocks, I'm pretty sure you'd end up with the exact same result.

No, the prev flag specifies how to deal with null blocks, prev=true says "walk backwards until the next non-null block" and prev=false say "walk forwards until the next non-null block"

Reply:
Yes but I need the state computed by executing `TS_B`. If I lookup `lookback+1` and take the parent state, I'll get the parent state of `TS_B`, not the state resulting from the evaluation of `TS_B`. The fundamental issue here is that the state root produced by executing `TS_B` isn't recorded anywhere on chain.

Reply:
(alternatively, we could have looked at the parent state of the lookback tipset, but that needs a network upgrade)

Reply:
What? if `a.Parents() == b` then `TipSetState(b) == a.ParentStateRoot`.

Reply:
> The fundamental issue here is that the state root produced by executing TS_B isn't recorded anywhere on chain.

Yes, its TS_C.ParentStateRoot

Reply:
Apparently I need to pay more attention when reading code. @whyrusleeping is correct.